src/ucm/malloc/malloc_hook.c

```diff
@@ -555,20 +555,27 @@ static void ucm_malloc_test(int events)
 
     ucm_event_handler_add(&handler);
 
-    /* Trigger both small and large allocations
-     * TODO check address / stop all threads */
-    for (i = 0; i < small_alloc_count; ++i) {
-        p[i] = malloc(small_alloc_size);
-    }
-    for (i = 0; i < small_alloc_count; ++i) {
-        free(p[i]);
-    }
-    p[0] = malloc(large_alloc_size);
-    p[0] = realloc(p[0], large_alloc_size * 2);
-    free(p[0]);
+    if (ucm_mmap_hook_mode() == UCM_MMAP_HOOK_RELOC) {
```

Comment:
in bistro mode we should not hook malloc calls.
calling malloc/free in bistro mode doesn't fire mmap events, just excluded it from processing due to minimize overhead

Reply:
why it does not fire mmap events? i think it should, since we are setting callbacks for them

Reply:
sometime it takes a huge number of mallocs to file brk/sbrk event, it may dramatically affect to startup time

Reply:
can we at least trigger mmap() by allocating large buffer?

Reply:
technically yes, but it will require knowledge of how glibc malloc works and how much free memory is requested from kernel. when I tried to implement such logic I found that amount of memory to allocate to fire event depends from available free memory on computer, how many allocations happened before (gtest sequence affected to this) and other magic which I didn't understood (sometime malloc fired event, but in most of times - not)

Reply:
ok then